diff --git a/src/dune b/src/dune
index 31106c84ff..dc31e5c493 100644
--- a/src/dune
+++ b/src/dune
@@ -49,7 +49,7 @@
   (public_name frama-c.kernel)
   (foreign_stubs (language c) (names c_bindings))
   (flags :standard -w -9)
-  (libraries frama-c.init fpath str unix zarith ocamlgraph dynlink bytes yaml.unix yojson menhirLib dune-site dune-site.plugins)
+  (libraries frama-c.init cudd fpath str unix zarith ocamlgraph dynlink bytes yaml.unix yojson menhirLib dune-site dune-site.plugins)
   (instrumentation (backend landmarks))
   (preprocess (staged_pps ppx_import ppx_deriving.eq ppx_deriving_yaml))
 )
diff --git a/src/kernel_services/ast_queries/cil_builtins.ml b/src/kernel_services/ast_queries/cil_builtins.ml
index a596fa02ee..336ece58fe 100644
--- a/src/kernel_services/ast_queries/cil_builtins.ml
+++ b/src/kernel_services/ast_queries/cil_builtins.ml
@@ -414,7 +414,7 @@ let init_builtins () =
     Kernel.fatal ~current:true "You must call initCIL before init_builtins" ;
   if Builtin_functions.length () <> 0 then
     Kernel.fatal ~current:true "Cil builtins already initialized." ;
-  init_builtins_from_json ();
+  let _ = init_builtins_from_json in
   Queue.iter (fun f -> register_custom_builtin (f())) custom_builtins
 
 (** This is used as the location of the prototypes of builtin functions. *)
--- a/src/libraries/utils/c_bindings.c	2023-10-10 04:03:10.009564948 +0200
+++ a/src/libraries/utils/c_bindings.c	2023-10-10 04:02:19.785242488 +0200
@@ -252,3 +252,5 @@
   double d = f;
   return caml_copy_double(d);
 }
+
+intnat caml_stat_compactions;
